File: app/src/main/java/sfllhkhan95/doodle/shapes/Pen.java
package sfllhkhan95.doodle.shapes;

import android.graphics.Paint;
import android.graphics.PointF;

import sfllhkhan95.doodle.core.PaintBrush;
import sfllhkhan95.doodle.core.PaintCanvas;

/**
 * Created by saifkhichi96 on 23/10/2017.
 */
public class Pen extends Shape {

    Pen(PaintBrush paintBrush) {
        super(paintBrush);
    }

    @Override
    public void paint(PaintCanvas canvas) {
        paintBrush.setStyle(Paint.Style.STROKE);
        paintBrush.setColor(paintBrush.getStrokeColor());
        paintBrush.setStrokeWidth(paintBrush.getSize());
        canvas.drawPath(this, paintBrush);
    }

    @Override
    public void draw(PointF i, PointF f) {
        this.lineTo(f.x, f.y);
    }
}

File: app/src/main/java/sfllhkhan95/doodle/models/shapes/Pen.kt
package sfllhkhan95.doodle.models.shapes

import android.graphics.Paint
import android.graphics.PointF

import sfllhkhan95.doodle.models.PaintBrush
import sfllhkhan95.doodle.models.PaintCanvas
import sfllhkhan95.doodle.models.Tool

/**
 * Created by saifkhichi96 on 23/10/2017.
 */
open class Pen internal constructor(paintBrush: PaintBrush) : Tool(paintBrush) {

    override fun paint(canvas: PaintCanvas) {
        paintBrush.style = Paint.Style.STROKE
        paintBrush.color = paintBrush.strokeColor
        paintBrush.strokeWidth = paintBrush.size.toFloat()
        canvas.drawPath(this, paintBrush)
    }

    override fun draw(i: PointF, f: PointF) {
        this.lineTo(f.x, f.y)
    }
}
